openedx/core/djangoapps/user_api/learner_skill_levels/utils.py

```diff
@@ -0,0 +1,65 @@
+from openedx.core.djangoapps.catalog.utils import (
+    get_catalog_api_client,
+    check_catalog_integration_and_get_user,
+    get_catalog_api_base_url,
+
+)
+from openedx.core.lib.edx_api_utils import get_api_data
+
+
+def get_top_skill_categories_for_job(job_id):
+    """
+        Retrieve top categories for the job with the given job_id.
+
+        Arguments:
+            job_id (int): id of the job about which we are retrieving information.
+
+        Returns:
+            dict with top 5 categories of specified job.
+    """
+    user, catalog_integration = check_catalog_integration_and_get_user(error_message_field='Skill Categories')
+    if user:
+        api_client = get_catalog_api_client(user)
+        root_url = get_catalog_api_base_url()
+        base_api_url = root_url.split('/api', 1)[0]
```

Comment:
This implementation looks a little hackish, Can we use parse_url to construct root_url?